### PR TITLE
[docs] Add regeneration section

### PR DESCRIPTION
**Pull request by Lewiky**

### CODE REVIEW

File: TempleEight/docs/guide/regeneration.md [hunk 1]
```diff
@@ -3,3 +3,84 @@ id: regeneration
 title: Regeneration
 sidebar_label: Regeneration
 ---
+
+In addition to generating code from a Templefile a single time to being project development, the Temple CLI also allows for regeneration of code.
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:7]: This first part doesn't make sense?

File: TempleEight/docs/guide/regeneration.md [hunk 2]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
+This means that if you generate a project, but then want to later add an additional service, or a property on that service, Temple will allow you 
+to regenerate code reflecting any changes to the original Templefile.
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:9]: should be on previous line

File: TempleEight/docs/guide/regeneration.md [hunk 3]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
 sidebar_label: Regeneration
 ---
+
+In addition to generating code from a Templefile a single time to being project development, the Temple CLI also allows for regeneration of code.
+This means that if you generate a project, but then want to later add an additional service, or a property on that service, Temple will allow you 
+to regenerate code reflecting any changes to the original Templefile. 
+
+During this process, Temple will regenerate _all_ project files, overwriting any existing ones, with a few key exceptions:
+
+- The _Setup_ source file, in any existing service
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:13]: Should `Setup` be capitalised?

File: TempleEight/docs/guide/regeneration.md [hunk 4]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
+
+- The _Datastore_ source file, in any existing service
+
+- Any file that belongs to a service that has been deleted from the Templefile since the last generation event.
+
+As the Setup and Datastore source files do not change based on the contents of the Service they're generated from, they can be safely omitted from the 
+regeneration process without risk of breaking changes.
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:20]: Should be on previous line

File: TempleEight/docs/guide/regeneration.md [hunk 5]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
+
+This also allows the user to add their own custom business logic (as per the [Business Logic](guides/business-logic) guide) in these two files, which persists
+between generation events.
+However it is very possible that the business logic that exists may not be correct after regeneration, reflecting changes to the properties of a service, etc.
+
+To invoke the regeneration process, simply run `temple generate` with an output directory that already contains _any of the files Temple is about to generate_.
+
+For example, first generate the project the first time:
+
+```
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:30]:
```suggestion
```bash
```

File: TempleEight/docs/guide/regeneration.md [hunk 6]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
+github.com/temple/tutorial
+The output directory already contains some files that are going to be generated.
+This process will overwrite them.
+These files will not be touched to preserve business logic:
+➤ example-service/dao/datastore.go
+➤ example-service/setup.go
+Do you want to continue? (Y/N)
+y
+Generated project in /Users/lewiky/Projects/TempleEight/temple-tutorial
+ ```
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:68]: This is dense, could we add some newlines?

File: TempleEight/docs/guide/regeneration.md [hunk 7]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
+
+ If we check the file we changed earlier, we'll see that the changes we made are still present.
+
+ ```
+ ~/Documents/temple-tutorial ❯❯❯ cat example-service/setup.go
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:73]: You've used `cat` here, but not in the above example? Can we avoid that so we can use go syntax highlighting

File: TempleEight/docs/guide/regeneration.md [hunk 8]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
+For example, first generate the project the first time:
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple generate example.temple
+What should the Go module name be? (expected format "github.com/username/repo")
+github.com/temple/tutorial
+Generated project in /path/Documents/temple-tutorial
+```
+
+Then, make some changes to a _Setup_ source file, we'll change the `example-service/setup.go` file:
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:37]: I'm not sure this makes sense either?

File: TempleEight/docs/guide/regeneration.md [hunk 9]
```diff
@@ -3,3 +3,84 @@ id: regeneration
...
+- Any file that belongs to a service that has been deleted from the Templefile since the last generation event.
+
+As the Setup and Datastore source files do not change based on the contents of the Service they're generated from, they can be safely omitted from the 
+regeneration process without risk of breaking changes. 
+
+This also allows the user to add their own custom business logic (as per the [Business Logic](guides/business-logic) guide) in these two files, which persists
+between generation events.
+However it is very possible that the business logic that exists may not be correct after regeneration, reflecting changes to the properties of a service, etc.
+
+To invoke the regeneration process, simply run `temple generate` with an output directory that already contains _any of the files Temple is about to generate_.
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:26]:
```suggestion
To invoke the regeneration process, simply run `temple generate` with an output directory that already contains an existing temple project.
```

File: TempleEight/docs/guide/regeneration.md (hunk 1 above)
Comment (Lewiky) [reply TempleEight/docs/guide/regeneration.md:7]: Lol I wrote that while also trying to focus on Dav-andy

File: TempleEight/docs/guide/regeneration.md [hunk 10]
```diff
@@ -3,3 +3,85 @@ id: regeneration
 title: Regeneration
 sidebar_label: Regeneration
 ---
+
+In addition to generating code from a Templefile a single time at the beginning of project development, the Temple CLI also allows for regeneration of code later in the cycle.
+This means that if you generate a project, but then want to later add an additional service, or a property on that service, Temple will allow you to regenerate code reflecting any changes to the original Templefile. 
+
+During this process, Temple will regenerate _all_ project files, overwriting any existing ones, with a few key exceptions:
```
Comment (xsanda) [inline TempleEight/docs/guide/regeneration.md:10]: I wonder if we should add a comment to the top of each file about whether they’re safe to edit?

File: TempleEight/docs/guide/regeneration.md [hunk 11]
```diff
@@ -3,3 +3,85 @@ id: regeneration
...
+
+This also allows the user to add their own custom business logic (as per the [Business Logic](guides/business-logic) guide) in these two files, which persists between generation events.
+However it is very possible that the business logic that exists may not be correct after regeneration, reflecting changes to the properties of a service, etc.
+
+To invoke the regeneration process, simply run `temple generate` with an output directory that already contains an existing temple project.
+
+For example, first generate the project the first time:
+
+```bash
+~/Documents/temple-tutorial ❯❯❯ temple generate example.temple
```
Comment (xsanda) [inline TempleEight/docs/guide/regeneration.md:28]: can we remove the path?

Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:28]:
```suggestion
❯❯❯ temple generate example.temple
```

File: TempleEight/docs/guide/regeneration.md [hunk 12]
```diff
@@ -3,3 +3,85 @@ id: regeneration
...
+func (env *env) setup(router *mux.Router) {
+	// Add user defined code here
+	fmt.Printf("Hello, World!\n")
+}
+```
+
+Next, we're going to regenerate the project, but we'll notice that the changes we just made will persist.
+
+```bash
+~/Documents/temple-tutorial ❯❯❯ temple generate example.temple
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:54]:
```suggestion
❯❯❯ temple generate example.temple
```

File: TempleEight/docs/guide/regeneration.md [hunk 13]
```diff
@@ -3,3 +3,85 @@ id: regeneration
...
+This process will overwrite them.
+
+These files will not be touched to preserve business logic:
+➤ example-service/dao/datastore.go
+➤ example-service/setup.go
+
+Do you want to continue? (Y/N)
+y
+
+Generated project in /Users/lewiky/Projects/TempleEight/temple-tutorial
```
Comment (jaylees14) [inline TempleEight/docs/guide/regeneration.md:69]: Probs just drop this to avoid confusion
```suggestion
```

**Pull request merged**